src/storage/src/manifest/impl_.rs

```diff
@@ -379,3 +382,46 @@ impl<S: Checkpoint<Error = Error>, M: MetaAction<Error = Error>> ManifestImplInn
         }
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct ImmutableManifestImpl<T> {
+    store: Arc<ImmutableManifestObjectStore>,
+    codec: Arc<dyn Codec<T, Error>>,
+}
```

Comment:
I think the ImmutableManifestImpl is over-designed. Currently, we don't need such a generic manifest and we have no restriction that we must put the manifest implementation in the storage crate.

In fact, we don't need to modify the storage crate. I prefer the most straightforward way: Implementing write/read helper functions in the `file-table-engine/manifest/immutable` mod (You could choose other names)
```rust
pub fn write_manifest(path: &str, object_store: &ObjectStore, metadata: &ImmutableMetadata) -> Result<()>;

pub fn read_manifest(path: &str, object_store: &ObjectStore) -> Result<Option<ImmutableMetadata>>;
```

We can call serde_json directly inside `read/write_manifest` and no need to define the Codec trait. So the total manifest implementation only requires about 100 lines of code.

Reply:
> In fact, we don't need to modify the storage crate. I prefer the most straightforward way: Implementing write/read helper functions in the `file-table-engine/manifest/immutable` mod (You could choose other names)


refactored 🚀